### PR TITLE
add myx adapter

### PR DESCRIPTION
**Pull request by marvin6en**

### CODE REVIEW

**Comment by 0xroll** [comment]
hi @marvin6en thanks for the pr.

have some question :
1) where can i find the codes that are used to build the subgraph?
2) does these cover the token balance changes in the lp pool due to traders' pnl ?
3) are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B

**Comment by nitish-91** [comment]
@marvin6en can you please follow readme to expose the function required, and use the input and output schema mentioned in it

**Comment by marvin6en** [comment]
> hi @marvin6en thanks for the pr.
> 
> have some question :
> 
> 1. can i have the github repo link for building the subgraph
> 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B

1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
2. pnl of users does not have an impact on the number of pool lp's
3. User deposits give the user lp tokens

**Comment by marvin6en** [comment]
> @marvin6en您可以按照自述文件公开所需的功能，并使用其中提到的输入和输出架构吗



> @marvin6en can you please follow readme to expose the function required, and use the input and output schema mentioned in it

I have confirmed that outputData.csv has been provided in the format you have requested

**Comment by 0xroll** [comment]
hey @marvin6en i mean the gh repo link

**Comment by marvin6en** [comment]
> hey @marvin6en i mean the gh repo link

It's not public yet.

**Comment by marvin6en** [comment]
hey @0xroll Is there anything else we need to modify? We want to merge this PR as soon as possible

**Comment by 0xroll** [comment]
> > hi @marvin6en thanks for the pr.
> > have some question :
> > 
> > 1. can i have the github repo link for building the subgraph
> > 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> > 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B
> 
> 1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
> 2. pnl of users does not have an impact on the number of pool lp's
> 3. User deposits give the user lp tokens, But right now we really aren't tracking the transfer of user lp tokens

actually we would need the transfers to be tracked. this captures the correct owner

also, any possibility of inviting me to your private subgraph repo?

**Comment by marvin6en** [comment]
> > > hi @marvin6en thanks for the pr.
> > > have some question :
> > > 
> > > 1. can i have the github repo link for building the subgraph
> > > 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> > > 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B
> > 
> > 
> > 
> > 1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
> > 2. pnl of users does not have an impact on the number of pool lp's
> > 3. User deposits give the user lp tokens, But right now we really aren't tracking the transfer of user lp tokens
> 
> actually we would need the transfers to be tracked. this captures the correct owner
> 
> also, any possibility of inviting me to your private subgraph repo?

1. If the transfer needs to be tracked, is it understandable to me that I need to give you all lp token holders

**Comment by marvin6en** [comment]
> > > > hi @marvin6en thanks for the pr.
> > > > have some question :
> > > > 
> > > > 1. can i have the github repo link for building the subgraph
> > > > 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> > > > 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B
> > > 
> > > 
> > > 
> > > 1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
> > > 2. pnl of users does not have an impact on the number of pool lp's
> > > 3. User deposits give the user lp tokens, But right now we really aren't tracking the transfer of user lp tokens
> > 
> > 
> > actually we would need the transfers to be tracked. this captures the correct owner
> > also, any possibility of inviting me to your private subgraph repo?
> 
> 1. If the transfer needs to be tracked, is it understandable to me that I need to give you all lp token holders

**Comment by marvin6en** [comment]
> > > > hi @marvin6en thanks for the pr.
> > > > have some question :
> > > > 
> > > > 1. can i have the github repo link for building the subgraph
> > > > 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> > > > 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B
> > > 
> > > 
> > > 
> > > 1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
> > > 2. pnl of users does not have an impact on the number of pool lp's
> > > 3. User deposits give the user lp tokens, But right now we really aren't tracking the transfer of user lp tokens
> > 
> > 
> > actually we would need the transfers to be tracked. this captures the correct owner
> > also, any possibility of inviting me to your private subgraph repo?
> 
> 1. If the transfer needs to be tracked, is it understandable to me that I need to give you all lp token holders
> 2. Please provide an account that requires permission, and I will give you permission for our private warehouse.

Hello, can you help me? @0xroll

**Comment by 0xroll** [comment]
> > > > > hi @marvin6en thanks for the pr.
> > > > > have some question :
> > > > > 
> > > > > 1. can i have the github repo link for building the subgraph
> > > > > 2. does these cover the token balance changes in the lp pool due to traders' pnl ?
> > > > > 3. are there lp tokens when user deposit into the pool? if yes, does it tracks token transfers? like user A provide LP -> transfer lp token to user B
> > > > 
> > > > 
> > > > 
> > > > 1. The subgraph url: https://subgraph-linea.myx.finance/subgraphs/name/myx-subgraph/graphql
> > > > 2. pnl of users does not have an impact on the number of pool lp's
> > > > 3. User deposits give the user lp tokens, But right now we really aren't tracking the transfer of user lp tokens
> > > 
> > > 
> > > actually we would need the transfers to be tracked. this captures the correct owner
> > > also, any possibility of inviting me to your private subgraph repo?
> > 
> > 
> > 
> > 1. If the transfer needs to be tracked, is it understandable to me that I need to give you all lp token holders
> > 2. Please provide an account that requires permission, and I will give you permission for our private warehouse.
> 
> Hello, can you help me? @0xroll

1) yes, the lp token represents who owns the liquidity, but if there is a staking/farming pool, then you need to get the "original" owner.

2) can you invite me to the gh repo for the subgraph?

**Comment by marvin6en** [comment]
@0xroll 
1. We have tracked the transfer of lp tokens in the subgraph, and now the data we provide is the latest holder
2. We have invited you to our private subgraph repo. You can take a look and let us know if you have any questions.

**Comment by 0xroll** [comment]
@marvin6en saw the repo.

looks like currently the adapter is returning the lp address and lp balance?

we will need them to be on underlying token asset level :

eg : btc - usdc pair -> btc and usdc separately.

and sum up these underlying token balance, to get the total balance of each asset.

token balances will need bigINT (raw number)

**Comment by nitish-91** [comment]
@marvin6en can you please close this pr and raise again. We have added a pr template which will help in reducing review back and forth

File: adapters/myx/src/sdk/positionSnapshots.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        }).then(res => res.json()) as SubgraphResponse
+        snapshotsArrays = snapshotsArrays.concat(res.data.userLPStats)
+        if (res.data.userLPStats.length !== 1000) {
+            break
+        }
+        skip += 1000
+        if (skip > 5000) {
+            skip = 0
+            b_start = snapshotsArrays[snapshotsArrays.length - 1].block + 1
+        }
```
[the remark this answers is not in the document]
Comment (marvin6en) [reply adapters/myx/src/sdk/positionSnapshots.ts:81]: This is the function of turning pages. If there is a large amount of data, you cannot obtain all the data at once.

**Pull request closed without merging**